mycroft/skills/media/vocab/__init__.py

```diff
@@ -0,0 +1,130 @@
+from os.path import join, dirname
+
+from adapt.intent import IntentBuilder
+from mycroft.skills import time_rules
+from mycroft.skills.core import MycroftSkill
+from mycroft.messagebus.message import Message
+from mycroft.configuration.config import ConfigurationManager
+
+from mycroft.util.log import getLogger
+import mopidy
+
+logger = getLogger(__name__)
+
+__author__ = 'forslund'
+
+
+class MediaSkill(MycroftSkill):
+    """
+        The MediaSkill class is a base class for media skills containing
+        vocabulary and intents for the common functions expected by a media
+        skill. In addition event handlers to lower volume when mycroft starts
+        to speak and raise it again when (s)he stops.
+
+        But wait there is one more thing! A small event emitter and matching
+        handler to stop media currently playing when new media is started.
+    """
+    def __init__(self, name):
+        super(MediaSkill, self).__init__(name)
+        self.isPlaying = False
+        config = ConfigurationManager.get_config()
+        self.base_conf = config.get('Media')
+
+    def initialize(self):
+        logger.info('Initializing MediaSkill commons')
+        logger.info('loading vocab files from ' + join(dirname(__file__),
+                    'vocab', self.lang))
+        self.load_vocab_files(join(dirname(__file__), 'vocab', self.lang))
+
+        self.register_vocabulary(self.name, 'NameKeyword')
+        self._register_common_intents()
+        self._register_event_handlers()
+
+    def _register_common_intents(self):
+        """
+           Register common intents, these include basically all intents
+           except the intents to start playback.
+        """
+        intent = IntentBuilder('NextIntent').require('NextKeyword')
+        self.register_intent(intent, self.handle_next)
+
+        intent = IntentBuilder('PrevIntent').require('PrevKeyword')
+        self.register_intent(intent, self.handle_prev)
+
+        intent = IntentBuilder('StopIntent').require('StopKeyword')
+        self.register_intent(intent, self.handle_stop)
+
+        intent = IntentBuilder('PauseIntent').require('PauseKeyword')
+        self.register_intent(intent, self.handle_pause)
+
+        intent = IntentBuilder('ResumeIntent').require('ResumeKeyword')
+        self.register_intent(intent, self.handle_resume)
```

Comment:
There's a couple of reasons
I have intentionally left all playIntents outside the base class. They will all be different, there might be several for each skill so I've left it for each skill to implement. Basically anything with "play" in it I've left for the inherited class to handle.

Pause/Resume was the latest additions so I haven't quite landed in how I to best implement it. A separate intent for resume was the most straightforward way without adding lots of logic to the existing skills.

I'm willing to do merge this into a PlayIntent if you'd like that better.